### PR TITLE
Fix: Upgraded Xen-tools from 4.15 to 4.19.0

### PR DESCRIPTION
**Pull request by roja-zededa**
1. Able to add ninja through apk in xen-tools/Docker and xen/Docker
2. Changed the seabios and Xen version to 1.16.3 and 4.19.0 respectively
3. Added 12-remove-vanillaqemu4.19-cpupinning.patch which removes new qemu_thread_set_affinity implementation (QEMU 8.0.4), also retained Nikolay CPU Pinning PatchNo.15
4. Warning treated as error flag is available by default so removed 12-disable-Werror-to-build-under-gcc-11.2.patch
5. NetdevTapOptions doesn't have has_br member so changed 10-bridge-helper-support.patch
6. Bydefault vhost-vsock and vhost-scsi is enabled so removing the corresponding enable flags from xen-tools/Dockerfile
7. Removed 11-char-socket-revert.patch as it's unnecessary
8. Removed [realtime] option from kvm.go and replaced it with [overcommit], hypervisor.go unit test need to be changed to reflect [overcommit]
9. 08-Revert__Revert__vfio_pci-quirks_c__Disable_stolen_memory_for_igd_VFIO__.patch looked super messy, so cleaned it. Apart from the above major changes, the index of other existing patches might have been changed.

### CODE REVIEW

File: pkg/xen-tools/Dockerfile [hunk 1]
```diff
@@ -26,10 +26,12 @@ ENV BUILD_PKGS \
     gettext iasl util-linux-dev ncurses-dev glib-dev \
     pixman-dev libaio-dev yajl-dev argp-standalone \
     linux-headers git patch texinfo tar libcap-ng-dev \
-    attr-dev flex bison cmake libusb-dev zstd-dev
+    attr-dev flex bison cmake libusb-dev zstd-dev \
+    python3 build-base ninja
```
Comment (rucoder) [inline pkg/xen-tools/Dockerfile:30]: ninja is replaced by samirai in Alpine. same binary different name https://pkgs.alpinelinux.org/package/edge/main/x86_64/samurai

File: pkg/xen-tools/Dockerfile [hunk 2]
```diff
@@ -57,11 +59,11 @@ RUN rm -rf /out/usr/man
...
 
 # When changing this version, also update SEABIOS_UPSTREAM_REVISION to whatever is in `Config.mk` in the xen source
-ENV XEN_VERSION 4.15.0
+ENV XEN_VERSION 4.19.0
 ENV XEN_SOURCE=https://downloads.xenproject.org/release/xen/${XEN_VERSION}/xen-${XEN_VERSION}.tar.gz
 ENV EXTRA_QEMUU_CONFIGURE_ARGS="--enable-libusb --enable-linux-aio \
-    --enable-vhost-net --enable-vhost-vsock --enable-vhost-scsi --enable-vhost-kernel \
-    --enable-vhost-user --enable-linux-io-uring --enable-zstd"
+--enable-vhost-net --enable-vhost-kernel \
+--enable-vhost-user --enable-linux-io-uring --enable-zstd"
```
Comment (rucoder) [inline pkg/xen-tools/Dockerfile:66]: @shjala didn't we replace io-uring with other lib yet?

File: pkg/xen-tools/Dockerfile [hunk 3]
```diff
@@ -89,10 +91,13 @@ RUN mkdir -p /out
 
 # FEATURES="--enable-stubdom --enable-vtpm-stubdom --enable-vtpmmgr-stubdom"
 COPY gmp.patch /xen/stubdom
+
 WORKDIR /xen
+
 # disable golang as it does not play well together with musl (stderr is defined as FILE* const and fails to compile)
 RUN ./configure --prefix=/usr --disable-xen --disable-golang --disable-qemu-traditional --disable-docs --enable-9pfs \
-                --with-system-ovmf=/usr/lib/xen/boot/ovmf.bin --disable-stubdom
+ --with-system-ovmf=/usr/lib/xen/boot/ovmf.bin --disable-stubdom \
```
Comment (rucoder) [inline pkg/xen-tools/Dockerfile:99]: I would keep original formating

File: pkg/xen-tools/Dockerfile (hunk 2 above)
Comment (shjala) [reply pkg/xen-tools/Dockerfile:66]: no, we did some test with @jsfakian and it turned out that other mechanism are slow in comparison.

Comment (OhmSpectator) [reply pkg/xen-tools/Dockerfile:66]: @shjala, @rucoder it's a wrong PR, go to #4133 =)

**Pull request closed without merging**